#64 Change routes to not include `userId`

lyio opened the pull request:
- Remove userId from routes
- Move repository out of controller to its own service

lyio committed:
#64 Change routes to not include `userId`

- Remove userId from routes
- Move repository out of controller to its own service

flashback2k14 commented:
please add a readme with instructions. can get it work!

@lyio 

<details>
<summar>
Error Log
==========
</summary>


[error] application - 

! @71ngk1nmf - Internal server error, for (GET) [/] ->
 
play.api.Configuration$$anon$1: Configuration error[Cannot connect to database [default]]
	at play.api.Configuration$.configError(Configuration.scala:154)
	at play.api.Configuration.reportError(Configuration.scala:806)
	at play.api.db.DefaultDBApi$$anonfun$connect$1.apply(DefaultDBApi.scala:48)
	at play.api.db.DefaultDBApi$$anonfun$connect$1.apply(DefaultDBApi.scala:42)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at play.api.db.DefaultDBApi.connect(DefaultDBApi.scala:42)
	at play.api.db.DBApiProvider.get$lzycompute(DBModule.scala:72)
	at play.api.db.DBApiProvider.get(DBModule.scala:62)
	at play.api.db.DBApiProvider.get(DBModule.scala:58)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
Caused by: play.api.Configuration$$anon$1: Configuration error[Failed to initialize pool: FATAL: database "socobo" does not exist]
	at play.api.Configuration$.configError(Configuration.scala:154)
	at play.api.PlayConfig.reportError(Configuration.scala:996)
	at play.api.db.HikariCPConnectionPool.create(HikariCPModule.scala:70)
	at play.api.db.PooledDatabase.createDataSource(Databases.scala:199)
	at play.api.db.DefaultDatabase.dataSource$lzycompute(Databases.scala:123)
	at play.api.db.DefaultDatabase.dataSource(Databases.scala:121)
	at play.api.db.DefaultDatabase.getConnection(Databases.scala:142)
	at play.api.db.DefaultDatabase.getConnection(Databases.scala:138)
	at play.api.db.DefaultDBApi$$anonfun$connect$1.apply(DefaultDBApi.scala:44)
	at play.api.db.DefaultDBApi$$anonfun$connect$1.apply(DefaultDBApi.scala:42)
Caused by: com.zaxxer.hikari.pool.HikariPool$PoolInitializationException: Failed to initialize pool: FATAL: database "socobo" does not exist
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:512)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:105)
	at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:71)
	at play.api.db.HikariCPConnectionPool$$anonfun$1.apply(HikariCPModule.scala:58)
	at play.api.db.HikariCPConnectionPool$$anonfun$1.apply(HikariCPModule.scala:54)
	at scala.util.Try$.apply(Try.scala:192)
	at play.api.db.HikariCPConnectionPool.create(HikariCPModule.scala:54)
	at play.api.db.PooledDatabase.createDataSource(Databases.scala:199)
	at play.api.db.DefaultDatabase.dataSource$lzycompute(Databases.scala:123)
	at play.api.db.DefaultDatabase.dataSource(Databases.scala:121)
Caused by: org.postgresql.util.PSQLException: FATAL: database "socobo" does not exist
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2458)
	at org.postgresql.core.v3.QueryExecutorImpl.readStartupMessages(QueryExecutorImpl.java:2589)
	at org.postgresql.core.v3.QueryExecutorImpl.<init>(QueryExecutorImpl.java:116)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:225)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:55)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:219)
	at org.postgresql.Driver.makeConnection(Driver.java:407)
	at org.postgresql.Driver.connect(Driver.java:275)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:95)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:101)
</details>

lyio committed:
#64 Change routes to not include `userId`

- Update the readme with information on how to run the application in dev and make H2 the default dev database

flashback2k14 commented:
- for me only _POST_ _/api/v1/recipes/ _ works! 
  - Why is there a backslash?
- the GET routes always return empty arrays
- inside Recipe.java on the categories property: intellij shows me this error message: 
  - "Element Collection" attribute value type should not be "Persistence Entity"
